### PR TITLE
[BE] Use `std::enable_if_t` in Math.h

### PR DESCRIPTION
**Pull request by malfet**
PyTorch is C++17 project, so let's use some C++17 convenience methods

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

File: aten/src/ATen/native/Math.h [hunk 1]
```diff
@@ -1235,7 +1235,7 @@ C10_UNUSED inline uint8_t abs_impl(uint8_t v) {
 }
 
 template <typename T>
-inline typename std::enable_if<std::is_integral<T>::value, T>::type
+inline typename std::enable_if_v<std::is_integral_t<T>, T>
```
Comment (Skylion007) [inline aten/src/ATen/native/Math.h:1238]:
```suggestion
inline typename std::enable_if_v<std::is_integral_v<T>, T>
```

File: aten/src/ATen/native/Math.h [hunk 2]
```diff
@@ -1235,7 +1235,7 @@ C10_UNUSED inline uint8_t abs_impl(uint8_t v) {
 }
 
 template <typename T>
-inline typename std::enable_if<std::is_integral<T>::value, T>::type
+inline typename std::enable_if_v<std::is_integral_v<T>, T>
```
Comment (malfet) [inline aten/src/ATen/native/Math.h:1238]:
```suggestion
inline typename std::enable_if_t<std::is_integral_v<T>, T>
```

File: aten/src/ATen/native/Math.h (hunk 1 above)
Comment (malfet) [reply aten/src/ATen/native/Math.h]: Sorry, silly typos, also `enable_if` should be `enable_if_t` :)

**Pull request closed without merging**